src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h

```diff
@@ -80,6 +80,16 @@ namespace OpenMS
       return db_;
     }
 
+    inline static ModificationsDB * getCustomInstance(OpenMS::String unimod_file, OpenMS::String psimod_file)
```

Comment:
I agree with Hannes that dropping the singleton pattern would be a good idea. After a quick grep to look at where ModificationsDB is used I see some possible (minor) issues in AASequence and ResidueDB and some other OpenMS classes. Here a ModificationsDB (reference/pointer/whatever) might need to be passed.